mantidproject/mantid
# Enable user-settable detector grouping angle in DirectILLReduction

This PR adds a new property to `DirectILLReduction`, `GroupingAngleStep`, which replaces the hard-coded 0.01 degrees (or was it radians?) step for detector grouping. Further, the grouping is now done using `GenerateGroupingPowder` which replaces the previous home-brewn solution.

**Report to:** [nobody].

**To test:**

You'll need to have the unit test data directory included in Mantid's data search directories.

Run the script below. See how the two output workspaces `sqw` and especially `stw` change when `GroupingAngleStep` is varied. 

```python
ws = DirectILLCollectData('ILL/IN4/084447.nxs')
DirectILLReduction(ws, OutputWorkspace='sqw', OutputSofThetaEnergyWorkspace='stw', GroupingAngleStep=5., Transposing='Transposing OFF')
```

Check that there won't be any leftover `.xml` or `.par` files in your temp directory afterwards.

Fixes #22223. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
